sys/uri_parser/uri_parser.c

```diff
@@ -38,7 +38,7 @@ static char *_consume_scheme(uri_parser_result_t *result, char *uri,
                              char *uri_end, bool *has_authority)
 {
     assert(uri);
-    assert(!has_authority);
+    assert(*has_authority == false);
```

Comment:
```suggestion
    assert(!(*has_authority));
```

Or is that too obscure?

Reply:
Alternatively, why not just set `has_authority` here? It is an internal function after all.

Reply:
I usually like it the way you suggested, but I thought putting the `== false` in there would make it a little simpler to understand

Reply:
>Alternatively, why not just set has_authority here? It is an internal function after all.

we could go with that. Included in the fixup.